### ISSUE TITLE
Cloud Storage: getStorageClass in SetStorageClassLifecycleAction is not visible

### ISSUE THREAD

**Issue by rmunna** [googleapis/google-cloud-java#7046]
getStorageClass() method in SetStorageClassLifecycleAction class is only visible inside the package. Its not accessible outside the package.

I am trying to read storage class set by the lifecycle rule through SDK in the following way. looks like its not possible. Can you suggest any other alternative to access if this is not possible.

f (lifecycleAction.getActionType().equals(SetStorageClassLifecycleAction.TYPE)) {
     return ((SetStorageClassLifecycleAction)lifeCycleRule.getAction()).getStorageClass(); 
**// Compiler error : The method getStorageClass() from the type BucketInfo.LifecycleRule.SetStorageClassLifecycleAction is  not visible**
}

#### Environment details

OS type and version: Windows 10
Java version: JDK 11
sdk version: google-cloud-storage-1.7.8.0.jar

**Comment by dmitry-fa** [comment]
Looks like a bug: getStorageClass() of BucketInfo.LifecycleRule.SetStorageClassLifecycleAction class should be declared as public.

To workaround this issue you can create a class in the same package:

```
package com.google.cloud.storage;

public class StorageClassExtractor {
    public static StorageClass extract(BucketInfo.LifecycleRule.LifecycleAction lifecycleAction) {
        return ((BucketInfo.LifecycleRule.SetStorageClassLifecycleAction)lifecycleAction).getStorageClass();
    }
}
```
Then you can use your helper class to access this method:
```
package storage;

import com.google.cloud.storage.BucketInfo;
import com.google.cloud.storage.StorageClass;
import com.google.cloud.storage.StorageClassExtractor;

public class LifecycleActionTest {
    public static void main(String... args) {
        BucketInfo.LifecycleRule.LifecycleAction lifecycleAction =
                BucketInfo.LifecycleRule.LifecycleAction.newSetStorageClassAction(StorageClass.MULTI_REGIONAL);
        if (lifecycleAction.getActionType().equals(BucketInfo.LifecycleRule.SetStorageClassLifecycleAction.TYPE)) {
            System.out.println(StorageClassExtractor.extract(lifecycleAction));
        }

    }
}
```

**Comment by rmunna** [comment]
> To workaround this issue you can create a class in the same package:
I don't get it.. Could you please elaborate.

when can I expect the bug fix on google-cloud-storage-1.7.8.0.jar?

**Comment by dmitry-fa** [comment]
> > To workaround this issue you can create a class in the same package:
> 
> I don't get it.. Could you please elaborate.
You can create StorageClassExtractor.java file in your project, it will have an access to the method your need. In your app you can use StorageClassExtractor to get StorageClass.

> 
> when can I expect the bug fix on google-cloud-storage-1.7.8.0.jar?

I will work on it, but I can't promise any date.

**Comment by rmunna** [comment]
workaround:

```
public StorageClass extractStorageClass(LifecycleAction lifecycleAction) {
    try {
      Method method = SetStorageClassLifecycleAction.class.getDeclaredMethod("getStorageClass");
      method.setAccessible(true);
      return (StorageClass)method.invoke(lifecycleAction);
    } catch (ReflectiveOperationException e) {
      throw new RuntimeException("Exception while reading storage class from lifecycle rules", e);
    }
  }
```

**Comment by dmitry-fa** [comment]
Yes, this is another workaround. But it will not work if you have security restrictions preventing you from invoking private methods. If you don't have such restrictions this will work fine.
